jina/executors/indexers/keyvalue.py

```diff
@@ -67,14 +69,45 @@ def add(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
             self.write_handler.body.write(value)
             self._size += 1
             # print(f'l: {l} p: {p} r: {r} r+l: {r + l} size: {self._size}')
+        self.write_handler.flush()
 
     def query(self, key: int) -> Optional[bytes]:
+        if self.query_handler is None:
+            self.query_handler = self.get_query_handler()
```

Comment:
Is this part necessary?

Reply:
I think this is not needed, but the problem is that the `test` is doing something that we are not doing. in the same context  managed indexer we are indexing, saving and querying

Reply:
It is not necessary and dangerous for now. Query handler and Write handler I think should not be opened at the same time.

Reply:
I think the reason refers to the comments I added in the test

Reply:
It was removed, but it a symptom of another problem we discussed over zoom. basically we need to flip the mutex and close handlers between ops. This is now done in the `.update`